qmp/libvirtgo_linux.go

```diff
@@ -0,0 +1,327 @@
+// Copyright 2016 The go-qemu Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package qmp
+
+import (
+	"errors"
+	"log"
+	"os"
+	"sync"
+	"time"
+
+	libvirt "github.com/rgbkrk/libvirt-go"
+)
+
+// LibvirtGoEventsInterval constant used by the getPollInterval
+// method.
+const LibvirtGoEventsInterval = "LIBVIRTGO_EVENTS_INTERVAL"
+
+type libvirtGoMonitorLinux struct {
+	LibvirtGoMonitor
+	domainName string
+	domain     *libvirt.VirDomain
+	uri        string
+	virConn    *libvirt.VirConnection
+	once       sync.Once
+	doneChan   chan bool
+	eventsChan chan Event
+}
+
+// NewLibvirtGoMonitor configures a connection to the provided hypervisor
+// and domain.
+// An error is returned if the provided libvirt connection URI is invalid.
+//
+// Hypervisor URIs may be local or remote, e.g.,
+//	qemu:///system
+//	qemu+ssh://libvirt@example.com/system
+func NewLibvirtGoMonitor(uri, domain string) Monitor {
```

Comment:
The other monitors in this package return a concrete type instead of an interface.  Let's stick with that for now for consistency.